subjects/java/checkpoints/factory-blueprint/README.md

```diff
@@ -0,0 +1,60 @@
+## Factory Blueprint
+
+### Instructions
+
+You are given an incomplete Factory design pattern implementation with some incorrect parts. Complete and fix the class to demonstrate your understanding of how the Factory design pattern works.
+
+### Expected Classes
+
+```java
+// Product interface
+public interface Product {
+    void showDetails();
+}
+
+// ConcreteProductA class
+public class ConcreteProductA {
+    ...
+}
+
+// ConcreteProductB class
+public class ConcreteProductB {
+    ...
+}
+
+// Factory class
+public class Factory {
+    public ConcreteProductB createProduct(String type) { // the type parametre accept two values `A` and `B`
```

Comment:
Shouldn't this be `public Product createProduct(...`?

Reply:
yes you are right